src/Raven.Server/Indexing/VoronIndexOutput.cs

```diff
@@ -85,21 +80,27 @@ public override void Seek(long pos)
             try
             {
                 base.Seek(pos);
-                _file.Seek(pos, SeekOrigin.Begin);
+                if (_ms != null)
+                    _ms.Seek(pos, SeekOrigin.Begin);
+                else
+                    _file.Seek(pos, SeekOrigin.Begin);
             }
             catch (IOException ioe) when (ioe.IsOutOfDiskSpaceException())
             {
                 ThrowDiskFullException();
             }
         }
 
-        public override long Length => _file.Length;
+        public override long Length => _ms?.Length ?? _file.Length;
 
         public override void SetLength(long length)
         {
             try
             {
-                _file.SetLength(length);
+                if(_ms != null)
+                    _ms.SetLength(length);
```

Comment:
what if that length exceeds the capacity?

Reply:
Cannot happen, this is used only to trim the file, not to extend it.
At any rate, that would increased the MemoryStream size.